src/engine/loopingcontrol.h

```diff
@@ -32,14 +32,14 @@ class LoopingControl : public EngineControl {
     // process() updates the internal state of the LoopingControl to reflect the
     // correct current sample. If a loop should be taken LoopingControl returns
     // the sample that should be seeked to. Otherwise it returns currentSample.
-    double process(const double dRate,
+    virtual double process(const double dRate,
```

Comment:
Does this mean `ReadAheadManager::addLoopingControl()` should take `EngineControl*` `LoopingControl*`? 

This way we loose the type safety and the nice `LoopingControl*` function preview in Eclipse. 
I rate this as more important than a diffuse fear for a non-propper stubs. Since LoopingControl is a relative independent object, it hardly lowers the test quality, if a not overridden function is called.